### ISSUE TITLE
ParseFragment "eating" elements

### ISSUE THREAD

**Issue by mganss** [AngleSharp/AngleSharp#358]
I'm trying to use the `ParseFragment` method to parse a fragment in the context of the body. When the fragment contains a `script` element, the element immediately after it seems to get lost:

``` C#
var parser = new HtmlParser();
var dom = parser.Parse("<html><body></body></html>");
var html = @"<script>alert('test');</script><p>Test</p>";
var nodes = parser.ParseFragment(html, dom.Body).ToArray();
// => nodes.Length == 1
// (HtmlScriptElement with InnerHtml == "<script>alert('test');Test</script>"
dom.Body.Append(nodes);
var s = dom.ToHtml();
// => s == "<html><head></head><body><script>alert('test');Test</script></body></html>"
```

This results in the same behavior:

``` C#
var dom = parser.Parse("<html><body></body></html>");
dom.Body.InnerHtml = html;
var s = dom.ToHtml();
// => s == "<html><head></head><body><script>alert('test');Test</script></body></html>"
```

This does not occur in a browser:

``` javascript
document.body.innerHTML = "<script>alert('test');</script><p>Test</p>";
// document.documentElement.outerHTML 
// === "<html><head></head><body><script>alert('test');</script><p>Test</p></body></html>"
```

**Comment by FlorianRappl** [comment]
Sounds like a bug to me. v0.9.7 is overdue since a long time, but this one should be addressed in it.

Thanks for reporting the issue! :+1:

**Comment by FlorianRappl** [comment]
Fixed, see PR #359. Will be included in the next release (I plan to roll it out on Sunday or Monday).
